### ISSUE TITLE
Question about multiple filters and file creation.

### ISSUE THREAD

**Issue by cogle** [GENIVI/dlt-daemon#302]
I currently have a config like so:

```
[FILTER1]
LogAppName=APP1
ContextName=.*
LogLevel=DLT_LOG_ERROR
File=TestDltLog
FileSize=5242880
NOFiles=300
[FILTER2]
LogAppName=APP2,APP3,APP4,APP5
ContextName=.*
LogLevel=DLT_LOG_INFO
File=TestDltLog
FileSize=5242880
NOFiles=300
```

When running sometimes I notice the following happen:

```
524221 Mar 19 08:38 TestDltLog_1_20210319-083628.dlt
524276 Mar 19 08:40 TestDltLog_2_20210319-083810.dlt
524239 Mar 19 08:42 TestDltLog_3_20210319-084023.dlt
114535 Mar 19 08:43 TestDltLog_4_20210319-084236.dlt
70270 Mar 19 08:43 TestDltLog_4_20210319-084237.dlt
```

To me this is unexpected, I would have thought that there should only be one TestDltLog_4_20210319-084237.dlt log.
It appears that a single app id is in the `TestDltLog_4_20210319-084237.dlt` and the other app id's are in the other ` TestDltLog_4_20210319-084236.dlt`.

I am using 2.18.4 is this possibly fixed in a later version, if so I will just update. Once this happens it begins to create many with the same name and different time stamp.

**Comment by cogle** [comment]
If it is not fixed could you point me to a possible quick and dirty patch that I might be able to implement. 
Thanks.

**Comment by cogle** [comment]
After reading the code I think this may fix the issue(https://github.com/GENIVI/dlt-daemon/commit/d49e8e3365dfc13c85506c0ff8686a708fd79f09), but feel free to correct me if I am wrong.

**Comment by thanhbnq** [comment]
Hello @cogle ,

This issue is fixed in the latest version. 
I have a quick check with the following configuration on my system:

```
[FILTER1]
LogAppName=LOG
ContextName=.*
LogLevel=DLT_LOG_ERROR
File=Test1
FileSize=7000
NOFiles=5
EcuID=ECU1

[FILTER2]
LogAppName=MT00,MT01
ContextName=.*
LogLevel=DLT_LOG_INFO
File=Test1
FileSize=7000
NOFiles=5
EcuID=ECU1
```

And I did not observe your reported issue!
Please kindly check at your side!
Thank you
Thanh

**Comment by thanhbnq** [comment]
> 
> 
> After reading the code I think this may fix the issue([d49e8e3](https://github.com/GENIVI/dlt-daemon/commit/d49e8e3365dfc13c85506c0ff8686a708fd79f09)), but feel free to correct me if I am wrong.

Yeah, it is actually a series of commits to fix the issue.
So the best way to go is please update to latest version!

In case you still face any more issues, please kindly let me know!

Regards,
Thanh

**Comment by cogle** [comment]
Thanks will update to the latest!
